x/distribution/abci.go

```diff
@@ -27,6 +27,13 @@ func BeginBlocker(ctx sdk.Context, k keeper.Keeper) error {
 		if err := k.AllocateTokens(ctx, previousTotalPower, ctx.CometInfo().LastCommit.Votes); err != nil {
 			return err
 		}
+
+		// every 1000 blocks send whole coins from decimal pool to community pool
+		if ctx.BlockHeight()%1000 == 0 {
```

Comment:
If Height == 0 (which should never happen), this check will also pass so let's just do this
```go
if h := ctx.BlockHeight(); h > 0 && h%1000 == 0 {
```

Reply:
I don't think we need that as L26 does already the check.

Reply:
can you provide context on why this is needed please?

Reply:
The decimal pool is getting only the scrums while the rest is send to the protocol pool. This checks every 1000blocks if the scrums are big enough to be sent.
Before if wasn't needed as the decimal pool was the community pool, so the scrums were just there as well.

Reply:
I guess my question is why are there "scrums" to begin with?

Reply:
We often use `TruncateDecimal`, for instance here: https://github.com/cosmos/cosmos-sdk/pull/18539/files#diff-1fba04fbe59b301da9da81cb18aa5b84f54840f2f4c396030e004c97f5e17ee9R252. Every remainder of truncate decimal is added to the decimal pool. There is decimal because we need to do ` DecCoins` -> ` Coins`